modules/apps/foundation/user-associated-data/user-associated-data-api/src/main/java/com/liferay/user/associated/data/model/impl/UADEntityAggregatorImpl.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.user.associated.data.model.impl;
+
+import com.liferay.user.associated.data.model.UADEntity;
+import com.liferay.user.associated.data.model.UADEntityAggregator;
+
+import java.util.List;
+
+/**
+ * @author William Newbury
+ */
+public abstract class UADEntityAggregatorImpl implements UADEntityAggregator {
```

Comment:
What is the purpose of this class? I'm worried that the way it is designed right now (specially due to the getUADEntities() method), this will not scale. We need to realize that there might be tons of entities of a give type and we cannot just load them all in memory for processing.